target/linux/ramips/image/mt7621.mk

```diff
@@ -637,6 +637,26 @@ define Device/hiwifi_hc5962
 endef
 TARGET_DEVICES += hiwifi_hc5962
 
+define Device/humax_e10
+  $(Device/dsa-migration)
+  $(Device/uimage-lzma-loader)
+  IMAGE_SIZE := 15680k
+  DEVICE_VENDOR := HUMAX
+  DEVICE_MODEL := E10
+  DEVICE_ALT0_VENDOR := HUMAX
+  DEVICE_ALT0_MODEL := QUANTUM E10
+  IMAGE/sysupgrade.bin := append-kernel | append-rootfs | \
+	edimax-header -s CSYS -m EA03 -f 0x70000 -S 0x01100000 | pad-rootfs | \
+	append-metadata | check-size
+  IMAGES += factory.bin
+  IMAGE/factory.bin := append-kernel | append-rootfs | pad-rootfs | \
+        edimax-header -s CSYS -m EA03 -f 0x70000 -S 0x01100000 | \
+        humax-factory f013c26cf0a320fb71d03356dcb6bb63 | check-size
```

Comment:
IMO check-size for a compressed image does not really make sense.
You could just drop it, or do check-size before the compression.

Reply:
Fixed!